lib/declarative_authorization/authorization.rb

```diff
@@ -276,36 +278,73 @@ def permit!(privilege, options = {})
-                  subject_type: "lease_document",
+                  subject_type: "lease_document"
                 )
-                actual_value = response[:subject_ids].any?
+                actual_value = response.any?
               end
 
               puts "  Is renewal - Rule requires: #{rule_requires}, Actual: #{actual_value}"
               if rule_requires == actual_value
-                condition_matched = true
                 puts "  ✓ Is_renewal condition matched!"
               else
                 puts "  ✗ Is_renewal condition did not match"
+                current_attribute_matched = false
+              end
+            end
+    
+            if conditions.key?(:id) && current_attribute_matched
+              condition_type = conditions[:id][0]
+              condition_proc = conditions[:id][1]
+    
+              if condition_type == :id_in_scope && condition_proc.is_a?(Proc)
+                puts "  Checking Occupancy ID in scope condition"
+                user_has_access_to_occupancy = attribute.validate?(attr_validator)
+                puts "  User has access to occupancy? #{user_has_access_to_occupancy}"
+                if user_has_access_to_occupancy
+                  puts "  ✓ This attribute matched conditions"
+                else
+                  puts "  ✗ This attribute did not match conditions"
+                  current_attribute_matched = false
+                end
               end
+            end
+
+            if current_attribute_matched
+              any_attribute_matched = true
+              puts "  ✓ All conditions matched for this attribute"
             else
-              puts "  (No is_renewal condition to check)"
+              all_attributes_matched = false
+              puts "  ✗ Not all conditions matched for this attribute"
             end
-            
-            if condition_matched
-              # For now, we will assume each rule's conditions are OR'd together. This is not the case
-              # for all rules, but it's the most common case and a good starting point.
-              puts "  At least one matching condition found, checking spicedb"
-              
+          end
+
+          puts "  Finished checking attributes, checking if rule is authorized"
+          
+          if rule.respond_to?(:join_operator) && rule.join_operator == :and
+            puts "  Rule uses AND operator, checking if all attributes matched: #{all_attributes_matched}"
+            if all_attributes_matched
+              puts "  All attributes matched, checking SpiceDB permission"
+              authorized = @auth_service.check_permission(
+                resource: { type: "vhost", id: vhost_id },
+                permission: permission_to_check
+              )
+              puts "  Authorized? #{authorized}"
+              return true if authorized
+            else
+              puts "  Not all attributes matched, authorization denied for this rule"
```

Comment:
Do we need to return false for this case or does it return false by default?

Reply:
We return `false` [here](https://github.com/appfolio/ae_declarative_authorization/pull/42/files#diff-98c137f16fa76b510685be5b02841a096937542cd920b8c80530d1a048872ccdR367) with logging, unless the method is called with `!`